src/Public/LogRhythm/Case/Update-LrCaseEarliestEvidence.ps1

```diff
@@ -0,0 +1,189 @@
+using namespace System.IO
+using namespace System.Collections.Generic
+
+Function Update-LrCaseEarliestEvidence {
+    <#
+    .SYNOPSIS
+        Update the earliest evidence timestamp of an existing case using a custom timestamp
+    .DESCRIPTION
+        The Update-LrCaseEarliestEvidenceFromDrilldown cmdlet updates an existing case's earliest evidence based on a timestamp.
+
+        Case Status must not be closed.
+    .PARAMETER Credential
+        PSCredential containing an API Token in the Password field.
+        Note: You can bypass the need to provide a Credential by setting
+        the preference variable $SrfPreferences.LrDeployment.LrApiToken
+        with a valid Api Token.
+    .PARAMETER Id
+        Unique identifier for the case, either as an RFC 4122 formatted string, or as a number.
+    .PARAMETER Timestamp
+        Timestamp in %format
+    .PARAMETER Quiet
+        Indicates that this cmdlet suppresses all output.
+    .INPUTS
+        [System.Object]     ->  Id
+        [System.String]     ->  Timestamp
+    .OUTPUTS
+        Optional summary output to validate Case Update status.
+    .EXAMPLE
+        PS C:\> Update-LrCaseEarliestEvidenceFromDrilldown -Id 8699 -Timestamp %TIME -Summary
+        ---
+        Updated Case: 8699 Based on Alarm: 396658 Drilldown Date: 2019-12-19T08:58:40Z
+        
+    .NOTES
+        LogRhythm-API
+    .LINK
+        https://github.com/SmartResponse-Framework/SmartResponse.Framework
+    #>
+
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory = $false, Position = 0)]
+        [ValidateNotNull()]
+        [pscredential] $Credential = $SrfPreferences.LrDeployment.LrApiToken,
+
+
+        [Parameter(
+            Mandatory = $true,
+            ValueFromPipelineByPropertyName = $true,
+            Position = 1
+        )]
+        [object] $Id,
+
+
+        [Parameter(Mandatory = $true, Position = 2)]
+        [ValidateNotNullOrEmpty()]
+        [string] $Timestamp,
+
+        [Parameter(Mandatory = $false, Position = 3)]
+        [switch] $PassThru,
+
+        [Parameter(Mandatory = $false, Position = 4)]
+        [switch] $Summary
+    )
+
+
+    Begin {
+        $Me = $MyInvocation.MyCommand.Name
+        
+        $BaseUrl = $SrfPreferences.LRDeployment.CaseApiBaseUrl
+        $Token = $Credential.GetNetworkCredential().Password
+
+        $ProcessedCount = 0
+    }
+
+
+    Process {
+        # Get Case Id
+        $IdInfo = Test-LrCaseIdFormat $Id
+        if (! $IdInfo.IsValid) {
+            throw [ArgumentException] "Parameter [Id] should be an RFC 4122 formatted string or an integer."
+        }
+
+
+        # Set Existing EarliestEvidence Date for comparison
+        $EarliestEvidence = Get-LrCaseEarliestEvidence -Id $Id
+        Write-Verbose "[$Me]: Case: $Id EarliestEvidence: $EarliestEvidence"
+
+        # Set Case Creation Date for comparison.  Earliest Evidence !> CaseCreationDate
+        $CaseCreate = (Get-LrCaseById -Id $Id).dateCreated
```

Comment:
My comment on `Get-LrCaseEarliestEvidence` recommended changing it's output from a `string` to a `datetime`.  This is a great example of why that will be good.   You won't have to convert it, and can immediately do the comparison with the `datetime` provided to this function.

Reply:
Updated to return the date/time in the datetime type.